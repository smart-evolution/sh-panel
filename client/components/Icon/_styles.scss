@font-face {
  font-family: 'icomoon';
  src: url('./fonts/icons.eot?vpbc3m');
  src:
    url('./fonts/icons.eot?vpbc3m#iefix') format('embedded-opentype'),
    url('./fonts/icons.ttf?vpbc3m') format('truetype'),
    url('./fonts/icons.woff?vpbc3m') format('woff'),
    url('./fonts/icons.svg?vpbc3m#icomoon') format('svg');
  font-weight: normal;
  font-style: normal;
}

.c-icon {
  /* stylelint-disable */
  font-family: 'icomoon';
  /* stylelint-enable */
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &--red {
    color: $color-roman;
  }

  &--green {
    color: $color-persian-green;
  }

  &--thermometer {
    &::before {
      content: "\e900";
    }
  }

  &--fire {
    &::before {
      content: "\e9a9";
    }
  }

  &--motion {
    &::before {
      content: "\e9dc";
    }
  }
}
