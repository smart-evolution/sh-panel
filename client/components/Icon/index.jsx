// @flow
import _ from 'lodash';
import React from 'react';

type Props = {
  type: string,
  className: string,
};

const Icon = (props: Props) => {
  const { type, className } = props;
  const typeClass = _.isEmpty(type) ? '' : `c-icon--${type}`;

  const classes = `c-icon ${typeClass} ${className}`;

  return <span className={classes} />;
};

Icon.defaultProps = {
  className: '',
};

export default Icon;
