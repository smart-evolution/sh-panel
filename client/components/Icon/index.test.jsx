import React from 'react';
import { mount } from 'enzyme';
import Icon from './index';

describe('Icon', () => {
  it('should render correctly', () => {
    const component = mount(<Icon />);

    expect(component).toMatchSnapshot();
  });
});
