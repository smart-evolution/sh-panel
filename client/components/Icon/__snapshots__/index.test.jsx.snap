// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<Icon
  className=""
>
  <span
    className="c-icon  "
  />
</Icon>
`;
